core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -200,7 +200,11 @@ public Path sharedDataFile() {
 
     /**
      * The data location with the cluster name as a sub directory.
+     *
+     * @deprecated Used to upgrade old data paths to new ones that do not include the cluster name, should not be used to write files to and
+     * will be removed in ES 6.0
      */
+    @Deprecated
```

Comment:
It's used by Security.java because we still need to add permissions for it (if it exists) so that we have permission to move the directory